### PR TITLE
feat: move `Loader` to `compat` package & replace with `BusyIndicator`

### PR DESCRIPTION
**Pull request by Lukas742**
BREAKING CHANGE: The deprecated `Loader` component has been moved to the `@ui5/webcomponents-react-compat` package. Please use the `BusyIndicator` instead, as it's now the only loading indicator that is supported by our UX guidelines. Please visit our [Migration Guide](https://sap.github.io/ui5-webcomponents-react/main/?path=/docs/migration-guide--docs) for more details.

### CODE REVIEW

File: packages/compat/src/components/Loader/Loader.stories.tsx [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-  title: 'User Feedback / Loader',
+  title: 'Loader',
   component: Loader,
   argTypes: {},
   args: {
     type: LoaderType.Indeterminate,
     progress: '60%'
-  }
+  },
+  tags: ['package:@ui5/webcomponents-compat']
```
Comment (MarcusNotheis) [inline packages/compat/src/components/Loader/Loader.stories.tsx:16]: Not a web component, so the tag should be removed

File: packages/compat/src/components/Loader/index.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { deprecationNotice, useI18nBundle, useStylesheet } from '@ui5/webcomponents-react-base';
 import { clsx } from 'clsx';
 import type { CSSProperties } from 'react';
 import { forwardRef, useEffect, useState } from 'react';
-import { LoaderType } from '../../enums/index.js';
-import { PLEASE_WAIT } from '../../i18n/i18n-defaults.js';
-import type { CommonProps } from '../../types/index.js';
+// todo: sb won't start with this (maybe because of vite.config alias paths?)
+// import { PLEASE_WAIT } from '@ui5/webcomponents-react/dist/i18n/i18n-defaults.js';
+import { PLEASE_WAIT } from '../../../../main/src/i18n/i18n-defaults.js';
```
Comment (MarcusNotheis) [inline packages/compat/src/components/Loader/index.tsx:10]: sounds like a vite config issue to be honest. Either we have to expose the i18n defaults in our main export, or add the config

File: packages/main/src/components/AnalyticalTable/index.tsx [hunk 3]
```diff
@@ -717,144 +730,145 @@ const AnalyticalTable = forwardRef<AnalyticalTableDomRef, AnalyticalTablePropTyp
...
-            {...getTableProps()}
-            tabIndex={showOverlay ? -1 : 0}
-            role="grid"
-            aria-rowcount={rows.length}
-            aria-colcount={visibleColumns.length}
-            data-per-page={internalVisibleRowCount}
-            data-component-name="AnalyticalTableContainer"
-            ref={tableRef}
-            className={tableClasses}
+        <BusyIndicator active={loading} data-component-name="AnalyticalTableBusyIndicator">
```
Comment (MarcusNotheis) [inline packages/main/src/components/AnalyticalTable/index.tsx:733]: should be support a `loadingDelay` prop as well?

File: packages/main/src/components/AnalyticalTable/index.tsx [hunk 4]
```diff
@@ -155,7 +169,6 @@ const AnalyticalTable = forwardRef<AnalyticalTableDomRef, AnalyticalTablePropTyp
     onSort,
     onTableScroll,
     onAutoResize,
-    LoadingComponent = DefaultLoadingComponent,
```
Comment (MarcusNotheis) [inline packages/main/src/components/AnalyticalTable/index.tsx:158]: remove from prop types

**Review by MarcusNotheis — CHANGES_REQUESTED** [review]

File: packages/main/src/components/AnalyticalTable/index.tsx (hunk 3 above)
Comment (Lukas742) [reply packages/main/src/components/AnalyticalTable/index.tsx:733]: Yup, but will be done in a different PR.

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**